client/src/main/java/io/spine/client/SubscribingRequest.java

```diff
@@ -0,0 +1,121 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.client;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import com.google.protobuf.Message;
+import io.grpc.stub.StreamObserver;
+import io.spine.base.MessageContext;
+
+import java.util.function.Consumer;
+
+/**
+ * Abstract base for client requests that subscribe to messages.
+ *
+ * @param <M>
+ *         the type of the subscribed messages
+ * @param <C>
+ *         the type of the context of messages or {@link io.spine.core.EmptyContext} if
+ *         messages do not have a context
+ * @param <W>
+ *         the type of the message that wraps a message and its context
+ *         (e.g. {@link io.spine.core.Event}); if subscribed message type does not have a context,
+ *         this parameter is likely to be the same as {@code M}
+ * @param <B>
+ *         the type of this requests for return type covariance
+ */
+public abstract class
+SubscribingRequest<M extends Message,
+                   C extends MessageContext,
+                   W extends Message,
+                   B extends SubscribingRequest<M, C, W, B>>
+        extends FilteringRequest<M, Topic, TopicBuilder, B> {
+
+    SubscribingRequest(ClientRequest parent, Class<M> type) {
+        super(parent, type);
+    }
+
+    abstract Consumers.Builder<M, C, W, ?> consumers();
+
+    abstract MessageConsumer<M, C> toMessageConsumer(Consumer<M> consumer);
+
+    @CanIgnoreReturnValue
+    public B observe(Consumer<M> consumer) {
+        consumers().add(toMessageConsumer(consumer));
+        return self();
+    }
+
+    /**
+     * Assigns a handler for the error reported to
+     * {@link StreamObserver#onError(Throwable)} of
+     * the {@link StreamObserver} responsible for delivering messages
+     * to the consumers.
+     *
+     * <p>Once this handler is called, no more messages will be delivered to consumers.
+     *
+     * @see #onConsumingError(ConsumerErrorHandler)
+     */
+    @CanIgnoreReturnValue
+    public B onStreamingError(ErrorHandler handler) {
+        consumers().onStreamingError(handler);
+        return self();
+    }
+
+    /**
+     * Assigns a handler for an error that may occur in the code of one of the consumers.
+     *
+     * <p>After this handler called, remaining consumers will get the message as usually.
+     *
+     * @see #onStreamingError(ErrorHandler)
+     */
+    @CanIgnoreReturnValue
+    public B onConsumingError(ConsumerErrorHandler<M> handler) {
+        consumers().onConsumingError(handler);
+        return self();
+    }
+
+    /**
+     * Creates and posts the subscription request to the server.
+     */
+    public Subscription post() {
+        Topic topic = builder().build();
+        StreamObserver<W> observer = createObserver();
+        return subscribe(topic, observer);
+    }
+
+    /**
+     * Subscribes to receive all messages of the specified type.
+     */
+    public Subscription all() {
```

Comment:
Why do we need this method? It looks like it does completely the same as `post()` but discards all previous filtering calls. But how would that be used in the client code?